### PR TITLE
feat: pdf export, fixes #3

### LINKED ISSUE #3: Export to pdf
Change graticule's source code in `.draw()` and `drawGrid` (etc.) to allow for exchanging the graphics context being worked on. Thereby allow for and actually implement the export of the diagram to PDF.

Library for PDF export: https://github.com/joshua-gould/canvas2pdf

### PR DESCRIPTION
**Pull request by AndyBuchwald**

### CODE REVIEW

**Comment by AndyBuchwald** [comment]
TODO: export button needs to be locked until analyze table is fully loaded

File: src/ui/analyze-popup.vue [hunk 1]
```diff
@@ -70,59 +30,21 @@
 <script>
 import { veil } from './veil.js'
 import PopupHeader from './popup-header.vue'
-import { mapState } from 'vuex'
+import AnalyzeTable from '../components/analyzeTable'
 import moment from 'moment'
+import { mapState } from 'vuex'
```
Comment (kinnarr) [inline src/ui/analyze-popup.vue:35]: Please remove the unused imports.

File: src/store/index.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import Vue from 'vue'
 import Vuex from 'vuex'
+import VueHtml2pdf from 'vue-html2pdf'
 import { Configuration } from '@/configuration'
 import metrics from '@/store/metrics'
 import AsyncComputed from 'vue-async-computed'
 
 Vue.use(Vuex)
+Vue.use(VueHtml2pdf)
 Vue.use(AsyncComputed)
```
Comment (kinnarr) [inline src/store/index.js:10]: I would prefer to have this in app.js or in a new file imported from app.js. Unlike Vuex, these plugins are not dedicated to the store.

File: src/store/index.js [hunk 3]
```diff
@@ -56,7 +58,8 @@ export default new Vuex.Store({
       state.configuration.exportFormat = newValue
     }
   },
-  actions: {},
+  actions: {
+  },
```
Comment (kinnarr) [inline src/store/index.js:62]: Please remove this if it is not from eslint

File: src/ui/export-popup.vue [hunk 4]
```diff
@@ -68,37 +68,66 @@
         <div class="modal-footer">
           <button
             class="btn btn-primary"
+            :disabled="!pdfReady"
```
Comment (kinnarr) [inline src/ui/export-popup.vue:71]: Do I also have to wait if I export an image without the table?

File: src/ui/export-popup.vue [hunk 5]
```diff
@@ -139,21 +179,75 @@ export default {
...
   },
   methods: {
-    doExport: function () {
-      const instance = window.MetricQWebView.instances[0]
-      instance.doExport()
+    doExport: async function () {
+      const canvas = this.createNewCanvas(document.getElementById('anaTable').offsetHeight)
+      this.addData(canvas)
+      this.image = canvas.toDataURL('image/png')
+      const result = await this.waitNewCanvas()
```
Comment (kinnarr) [inline src/ui/export-popup.vue:186]: Is this code only for the pdf export or for all exports?

**Review by kinnarr — CHANGES_REQUESTED** [review]
Looks good overall. Just a few comments about code placement

File: src/store/index.js [hunk 6]
```diff
@@ -58,8 +54,6 @@ export default new Vuex.Store({
       state.configuration.exportFormat = newValue
     }
   },
-  actions: {
-  },
```
Comment (kinnarr) [inline src/store/index.js]: Sorry, I just meant the line break.

**Review by kinnarr — CHANGES_REQUESTED** [review]
Please rebase and fix the conflict in store.js

**Review by kinnarr — APPROVED** [review]
LGTM

**Pull request merged**